core/src/main/scala/akka/persistence/r2dbc/ConnectionFactoryProvider.scala

```diff
@@ -45,68 +44,23 @@ class ConnectionFactoryProvider(system: ActorSystem[_]) extends Extension {
         .map(_ => Done)
     }
 
-  def connectionFactoryFor(configLocation: String): ConnectionFactory = {
+  def connectionFactoryFor(settings: R2dbcSettings, configLocation: String): ConnectionFactory = {
```

Comment:
I think this is somewhat weird. ConnectionFactoryProvider is an extension so that projections can use the same connection pool, or users can use the same connection pool with R2dbcSession. Here we pass in R2dbcSettings, which is not part of the key, so it's only used once for the first call of a given configLocation. Subsequent calls may have different R2dbcSettings.

See also https://doc.akka.io/docs/akka-projection/current/r2dbc.html#multiple-plugins

can't we place everything needed for the ConnectionFactory in the same place, i.e.

```
akka.persistence.r2dbc.postgres.connection-factory {
}
```
which is the configLocation here.

That can even include a dialect property so that we can create the Dialect also from here

Reply:
The tricky thing is that the connection for h2 needs the init with db schema if we want to be sure to not start using it before the tables exist, and to create the schema we needs any overrides to the table names which are in the surrounding settings.

Especially tricky for projections because there we want the schema to include projection tables together with journal if both are used, but should also work with only projection running h2 and not using the journal at all (creating the tables but not using is perhaps fine though).

Reply:
I'll anyway try this idea out, I can't think of another possible solution.